instrumentation/src/main/java/io/opentelemetry/android/features/diskbuffering/DiskBufferingConfiguration.java

```diff
@@ -5,16 +5,22 @@
 
 package io.opentelemetry.android.features.diskbuffering;
 
+import io.opentelemetry.android.features.diskbuffering.scheduler.DefaultExportScheduleHandler;
+import io.opentelemetry.android.features.diskbuffering.scheduler.DefaultExportScheduler;
+import io.opentelemetry.android.features.diskbuffering.scheduler.ExportScheduleHandler;
+
 /** Configuration for disk buffering. */
 public final class DiskBufferingConfiguration {
     private final boolean enabled;
     private final int maxCacheSize;
+    private final ExportScheduleHandler exportScheduleHandler;
```

Comment:
I think this is problematic because it introduces an action class as a field of a configuration data class. It mixes up dumb/simple config with things that do work....and I think that's a troublesome path. 

What are the use cases that you can think of for allowing a user to pass in their own `ExportScheduleHandler` instance to the `DiskBufferingConfiguration`? I thought that maybe it was to specify their own schedule/period, but the way the code is structured now I'm not even sure that's possible.

If that was the main intent, I'd rather that the period be the configurable thing, and the builders for the action classes can do the building by using the config, not by having the config do the building itself. Make sense?

Reply:
> What are the use cases that you can think of for allowing a user to pass in their own `ExportScheduleHandler` instance to the `DiskBufferingConfiguration`?

The default "scheduler" only works while the app is running and I'm guessing some users would like to attempt exporting data even if their app isn't running by using  [WorkManager](https://developer.android.com/develop/background-work/background-tasks/persistent) for example.

Though I agree that the config should generally be made of data instead of functions or the like, maybe since the disk buffering feature is needed to initialize RUM, we could categorize it as an "initialization" feature, which will allow us to get rid of `DiskBufferingConfiguration` altogether as we would only use `OtelRumConfig` to set all initialization params, and then we could move the ExportScheduleHandler setter over to `OpenTelemetryRumBuilder` if we decided to go with the proposed structure in [here](https://github.com/open-telemetry/opentelemetry-android/issues/214).

Reply:
I'm ok with moving this around after this PR, possibly as part of dealing with the larger config structure questions.